src/UI/NavigationContainer.elm

```diff
@@ -173,6 +173,7 @@ type alias Container msg =
     , title : String
     , dialog : Maybe (Dialog msg)
     , hasMenu : Bool
+    , hasPersistentSidebar : Bool
```

Comment:
this change will break in a lot of places, is there any alternative? If not then I suggest we include a record with the defaults so that next time including a new option doesn't break everything

Reply:
the whole `UI.NavigationContainer` should be rewritten, having `Sidebar` and `Dialog` as separated modules is a good beginning.

This record, for example, could fully become a with-pattern composable type...

Could you create a DES card for it?

Reply:
Done, since I think you're more acquainted with the module I'll leave it for you to add a description. This might be something worth sketching an RFC before jumping directly into a refactor.